### PR TITLE
Add generation of constant themes

### PR DESCRIPTION
**Pull request by jakubtiteo**

### CODE REVIEW

File: packages/theme_tailor/lib/src/generator/theme_tailor_generator.dart [hunk 1]
```diff
@@ -178,6 +220,17 @@ class _TailorClassVisitor extends SimpleElementVisitor {
     if (ignoreAnnotationTypeChecker.hasAnnotationOf(element)) return;
 
     if (element.isStatic && element.type.isDartCoreList) {
+      if (!element.isConst) {
+        hasNonConstantElement = true;
+
+        if (requireConstThemes) {
+          print(
+              'Field "${element.name}" needs to be a const in order to be included'
+              ' in generated theme');
```
Comment (Rongix) [inline packages/theme_tailor/lib/src/generator/theme_tailor_generator.dart:229]: Actually we should throw an error here since it is required

File: packages/theme_tailor/lib/src/generator/theme_tailor_generator.dart [hunk 2]
```diff
@@ -40,36 +41,66 @@ class ThemeTailorGenerator extends GeneratorForAnnotation<Tailor> {
         todo: 'Move @Tailor annotation above `class`',
       );
     }
-    final library = element.library;
+    final resolver = buildStep.resolver;
+    final library = await resolver
+        .libraryFor(await resolver.assetIdForElement(element.library));
+
```
Comment (Rongix) [inline packages/theme_tailor/lib/src/generator/theme_tailor_generator.dart:47]: Is this change required? (I didn't catch why it is necessary)

Comment (jakubtiteo) [reply packages/theme_tailor/lib/src/generator/theme_tailor_generator.dart]: I tested the code generation a lot of times, and once in a while I was getting exception originating from this line, but I don't really remember what that an exception was because I worked on this long time ago. I will remove this change for now since it is not really related to constant themes.

File: packages/theme_tailor/README.md [hunk 3]
```diff
@@ -240,7 +241,31 @@ class NotGeneratedExtension extends ThemeExtension<NotGeneratedExtension> {
 }
 ```
 
-To see examle implementation of nested theme, head out to: [example:nested_themes][example: nested_themes]
+To see example implementation of nested theme, head out to: [example:nested_themes][example: nested_themes]
```
Comment (PatrykPawlak) [inline packages/theme_tailor/README.md:244]: Current markdown tag is inappropriate, as display only text, not link to example, so could be fixed by the way (without additional PR)
```suggestion
To see example implementation of nested theme, head out to: [example: nested_themes][example:nested_themes]
```

**Review by RSC-88 — APPROVED** [review]

**Review by Rongix — APPROVED** [review]

**Review by Axot017 — APPROVED** [review]
